CHANGELOG.md

```diff
@@ -4,6 +4,162 @@ All notable changes to Router will be documented in this file.
+
+As another example, this logs the bodies of requests:
+
+```rust
+// Generate a log for each request at this stage
+fn process_request(request) {
+    print(`body: ${request.body}`);
+}
+
+// Generate a log for each response at this stage
+fn process_response(response) {
+    print(`body: ${response.body}`);
+}
+```
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/3234
+
+## 🐛 Fixes
+
+### Small performance improvements to telemetry ([PR #3656](https://github.com/apollographql/router/pull/3656))
+
+We applied some small performance improvements to the `SpanMetricsExporter` (which is used to report span timings), some of which apply in cases where telemetry is disabled and could be apparent to most users.
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/3656
+
+### Handle interfaces in fragment spreads when `__typename` is omitted ([Issue #2587](https://github.com/apollographql/router/issues/2587))
+
+We now check the parent type when using an inline-fragment, rather than relying on the expectation that `__typename` will be present, since that is not a guarantee.  For cases where `__typename` was being omitted, this corrects cases where a portion of the selection set was silently dropped and not returned.
+
+By [@o0Ignition0o](https://github.com/o0Ignition0o) and [@geal](https://github.com/geal) in https://github.com/apollographql/router/pull/3718
+
+### Deduplication is, again, enabled by default as documented ([PR #3773](https://github.com/apollographql/router/pull/3773))
+
+[Subscription deduplication](https://www.apollographql.com/docs/router/executing-operations/subscription-support#subscription-deduplication) is again enabled by default as it was intended to be.  This important performance feature for subscriptions at scale was inadvertently disabled in v1.25.0 due to a bug.
+
+Should you still wish to disable it, you can still explicitly [set `enable_deduplication` to `false` in your configuration](https://www.apollographql.com/docs/router/executing-operations/subscription-support/#disabling-deduplication).
+
+By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/3773
+
+### Metrics are no longer coerced incorrectly ([Issue #3687](https://github.com/apollographql/router/issues/3687))
+
+Metric attributes are no longer incorrectly coerced to strings.  In addition, the logic around types which are accepted as metrics attributes has been simplified to avoid this in the future.  Going forward, if the wrong type is specified, values will be ignored and a log message will be emitted.
```

Comment:
```suggestion
Metric attributes are no longer incorrectly coerced to strings.  In addition, the logic around types which are accepted as metrics attributes has been simplified to avoid this in the future.  Going forward, if the wrong type is specified, values will be ignored and a log message (at debug level) will be emitted.
```